File: Java/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
       if (strs.length == 0 || strs[0].length() == 0) {
            return "";
        }
        int i = 0;
        while (i < strs[0].toCharArray().length) {
            for (int j = 1; j < strs.length; j++) {
                if (strs[j].length() < i + 1 || strs[0].toCharArray()[i] != strs[j].toCharArray()[i]) {
                    return strs[0].substring(0, i);
                }
            }
            i++;
        }
        return strs[0];
    }
}

File: Kotlin/longest-common-prefix.kt
class Solution {
    fun longestCommonPrefix(strs: Array<String>): String {
        if (strs.size == 0 || strs[0].length == 0) {
            return ""
        }
        for (i in 0 until strs[0].length) {
            for (j in 1 until strs.size) {
                if (strs[j].length < i + 1 || strs[0][i] != strs[j][i]) {
                    return strs[0].take(i)
                }
            }
        }
        return strs[0]
    }
}
